### DISCUSSION TITLE
CLI configuration not working?

### DISCUSSION THREAD

**Discussion by robertoranon** [donmccurdy/glTF-Transform#854] [Q&A] [answered]
Hi,

   I am new to glTF-Transform, and I would like to use it for a client project. The first thing I would like to do is a CLI tool that performs specific modifications of certain types of material, so I thought I'd use CLI configuration, i.e. extend the already functioning CLI.

The docs page at https://gltf-transform.donmccurdy.com/cli.html says that I should provide a configuration mis file, which, in my case, is:

```
// Custom transform: convert materials
function convertMaterials(options) {
    return (document) => {
        for (const material of document.getRoot().listMaterials()) {
            console.log(material);
        }
    };
}
```

however, when I try to install the custom command `convertMaterials` with

`gltf-transform --help --config path/to/custom-transform.config.mjs`

(of course using the right path):
- first, I do not get any feedback about the command being installed 
- second, if I run `gltf-transform convertMaterials model.glb` 

I get error: Unknown command convertMaterials

Am I doing something wrong?

Thanks for any help!

**Comment by donmccurdy** [comment, marked as answer]
Hi @robertoranon! You'll need to export a configuration definition, glTF Transform cannot detect new commands from the function declaration alone. Something like this should work:

```javascript
// Custom transform: convert materials
function convertMaterials(options) {
    return (document) => {
        for (const material of document.getRoot().listMaterials()) {
            console.log(material);
        }
    };
}

export default {
 onProgramReady: ({ program, io, Session }) => {
     // Usage: https://caporal.io/
     program
         .command('convertMaterials', 'Convert materials')
         .help('Lorem ipsum dolorem...')
         .argument('<input>', 'Path to read glTF 2.0 (.glb, .gltf) model')
         .argument('<output>', 'Path to write output')
         .action(({ args, options, logger }) =>
             Session.create(io, logger, args.input, args.output).transform(convertMaterials(options))
         );
 },
};
```

**Comment by robertoranon** [comment]
@donmccurdy thanks, it works.

For anyone coming to the same issue: after preparing the .mjs file as Don suggests, you need to call it as follows:

`gltf-transform convertMaterials input.glb output.glb  --config path/to/convert-materials.mjs`

From the docs I somehow understood that I had to call a command to `install` the custom command, and then I could use it, but that's not the case

**Reply by donmccurdy** [reply]
At some point I may have glTF Transform look for the config file automatically, but for now it does need to be specified with every call, yes. 👍🏻
